pkg/controllers/hub/membercluster/controlller_integration_test.go

```diff
@@ -0,0 +1,188 @@
+	Context("Test MemberCluster controller, handle force delete", func() {
+		BeforeEach(func() {
+			mc := clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: memberClusterName,
+					// finalizer is added to ensure MC is not deleted before the force delete wait time,
+					// ideally added and removed by fleet hub member cluster controller.
+					Finalizers: []string{"test-member-cluster-cleanup-finalizer"},
+				},
+				Spec: clusterv1beta1.MemberClusterSpec{
+					Identity: rbacv1.Subject{
+						Name:      "test-subject",
+						Kind:      "ServiceAccount",
+						Namespace: "fleet-system",
+						APIGroup:  "",
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, &mc)).Should(Succeed())
+
+			// Create the fleet member namespace.
+			memberNS := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fleetMemberNS,
+				},
+			}
+			Expect(hubClient.Create(ctx, &memberNS)).Should(Succeed())
+
+			// Create the EndpointSliceImports.
+			for i := range endpointSliceImportNames {
+				esi := buildEndpointSliceImport(endpointSliceImportNames[i])
+				Expect(hubClient.Create(ctx, esi)).Should(Succeed())
+			}
+		})
+
+		It("should remove finalizer on EndpointSliceImport on fleet member namespace, after force delete wait time is crossed", func() {
+			// ensure EndpointSliceImports have a finalizer.
+			var esi fleetnetv1alpha1.EndpointSliceImport
+			for i := range endpointSliceImportNames {
+				Expect(hubClient.Get(ctx, types.NamespacedName{Name: endpointSliceImportNames[i], Namespace: fleetMemberNS}, &esi)).Should(Succeed())
+				Expect(esi.GetFinalizers()).ShouldNot(BeEmpty())
+			}
+			// delete member cluster to trigger member cluster controller reconcile.
+			var mc clusterv1beta1.MemberCluster
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: memberClusterName}, &mc)).Should(Succeed())
+			Expect(hubClient.Delete(ctx, &mc)).Should(Succeed())
+			// the force delete wait time is set to 1 minute for this IT.
+			Eventually(func() error {
+				var endpointSliceImportList fleetnetv1alpha1.EndpointSliceImportList
+				if err := hubClient.List(ctx, &endpointSliceImportList, client.InNamespace(fleetMemberNS)); err != nil {
+					return err
+				}
+				if len(endpointSliceImportList.Items) != len(endpointSliceImportNames) {
+					return fmt.Errorf("length of listed endpointSliceImports %d doesn't match length of endpointSliceImports created %d",
+						len(endpointSliceImportList.Items), len(endpointSliceImportNames))
+				}
+				for i := range endpointSliceImportList.Items {
+					esi := &endpointSliceImportList.Items[i]
+					if len(esi.GetFinalizers()) != 0 {
+						return fmt.Errorf("finalizers on EndpointSliceImport %s/%s have not been removed", esi.Namespace, esi.Name)
+					}
+				}
+				return nil
+			}, eventuallyTimeout, eventuallyInterval).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			// Delete the namespace, the namespace controller doesn't run in this IT
```

Comment:
Similar pattern https://github.com/Azure/fleet/blob/main/pkg/controllers/membercluster/v1beta1/membercluster_controller_integration_test.go#L662-L664

Reply:
yep, I asked the controller runtime to add it and they refused.